### PR TITLE
Support trustedTesters.

### LINKED ISSUE #26: How do i publishing it to trusted testers?
```
Response code 400 (Bad Request)
{
    "error": {
        "errors": [
            {
                "domain": "global",
                "reason": "badRequest",
                "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
            }
        ],
        "code": 400,
        "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
    }
}
```

### PR DESCRIPTION
**Pull request by squidarth**
Hi there!

Thanks a lot for making this great CLI tool! In this PR, I added a `trusted-testers` option to the CLI to be used with the `publish` command so that we can use this to support publishing to trusted testers only.

Let me know what you think!

Fixes #26

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -103,7 +106,8 @@ if (isUpload) {
 
 if (isPublish) {
     spinnerStart('Publishing');
-    publish({ apiConfig }).then(res => {
+
+    publish({ apiConfig }, trustedTesters ? "trustedTesters" : undefined ).then(res => {
```
Comment (DrewML) [inline index.js]: Mind changing this to just:

```js
publish({ apiConfig }, trustedTesters && "trustedTesters").then(res => {
```

`trustedTesters` should already be `undefined` if not passed in.

Happy to merge/release once you make that change. Really appreciate your contribution!

**Comment by squidarth** [comment]
@DrewML Updated! Thanks for looking at this so quickly

**Comment by powdahound** [comment]
Would love to make use of this improvement if you have time to merge this and cut a new build. Thanks!

**Comment by sindresorhus** [comment]
This needs some documentation.

**Comment by squidarth** [comment]
@sindresorhus Hey, sorry about the long delay on this. Added the option to `README.md`, let me know if there's anything else you had in mind here

File: index.js [hunk 2]
```diff
@@ -27,6 +27,8 @@ const cli = meow(`
       --client-secret   OAuth2 Client Secret
       --refresh-token   OAuth2 Refresh Token
       --auto-publish    Can be used with the "upload" command
+      --trusted-testers    Can be used with the "publish" command
```
Comment (sindresorhus) [inline index.js]: The other descriptions need to be aligned with this one, like in the readme.

Comment (squidarth) [inline index.js]: cool--fixed

**Comment by p0we7** [comment]
if i used upload and auto-publish these will still response error 400 likly #26

**Comment by squidarth** [comment]
@sindresorhus @DrewML Any other changes you'd like to see before merging this?

**Comment by bfred-it** [comment]
I'd like to see this tested by someone who uses `trustedUsers`. @p0we7 seems to be having issues with it

**Comment by bfred-it** [comment]
@greghuc @p0we7 can you try this branch?

```sh
npm i -g finventures/chrome-webstore-upload-cli
```

**Comment by p0we7** [comment]
@bfred-it  still have error,  already uploaded extension zip, but can't auto-publish

```
root@ubuntu:~# npm i -g finventures/chrome-webstore-upload-cli
/usr/bin/webstore -> /usr/lib/node_modules/chrome-webstore-upload-cli/index.js
/usr/lib
└─┬ chrome-webstore-upload-cli@1.1.1  (git://github.com/finventures/chrome-webstore-upload-cli.git#389cef311e74220e47748ac70b2691a8a56d7ca0)
  ├─┬ ...........
  

root@ubuntu:~# npm list -g --depth=0
/usr/lib
├── chrome-webstore-upload-cli@1.1.1 (git://github.com/finventures/chrome-webstore-upload-cli.git#389cef311e74220e47748ac70b2691a8a56d7ca0)
└── npm@3.10.10

root@ubuntu:~/myproject# export CLIENT_ID="000000000000-oooooooooooooooooooooooooooooooo"
root@ubuntu:~/myproject# export CLIENT_SECRET="oooooooooooooooooooooooo"
root@ubuntu:~/myproject# export REFRESH_TOKEN="ooooooooooooooooooooooooooooooooooooooooooooo"
root@ubuntu:~/myproject# 
root@ubuntu:~/myproject# webstore upload --extension-id nnoooooooooooooooooooooooooooooo --trusted-testers --auto-publish
Response code 400 (Bad Request)
{
    "error": {
        "errors": [
            {
                "domain": "global",
                "reason": "badRequest",
                "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
            }
        ],
        "code": 400,
        "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
    }
}
root@ubuntu:~/myproject# ^C
root@ubuntu:~/myproject# 

```

**Comment by bfred-it** [comment]
@p0we7 thanks!

For some reason the API seems to require an additional `publish_to_trusted_testers=true` parameter to the request, even if it's not documented anywhere [[1]](https://developer.chrome.com/webstore/webstore_api/items/publish#parameters) [[2]](https://developer.chrome.com/webstore/using_webstore_api). The module [chrome-extension-deploy](https://github.com/erikdesjardins/chrome-extension-deploy) seems to do this: 

https://github.com/erikdesjardins/chrome-extension-deploy/blob/68c3d48614f69f762e4a6d91f6cf157338b8192b/index.js#L82-L83

**Comment by bfred-it** [comment]
This requires some changes in the chrome-webstore-upload package. I pushed some changes to its repo on a separate branch. @p0we7 try reinstalling it

```sh
npm rm -g finventures/chrome-webstore-upload-cli && npm i -g finventures/chrome-webstore-upload-cli
```

**Comment by p0we7** [comment]
I am sorry that the problem still exists. 
And i looks like this source does not have debug option
@bfred-it 


```
root@ubuntu:~/myproject# npm rm -g finventures/chrome-webstore-upload-cli && npm i -g finventures/chrome-webstore-upload-cli
/usr/bin/webstore -> /usr/lib/node_modules/chrome-webstore-upload-cli/index.js
/usr/lib
└─┬ chrome-webstore-upload-cli@1.1.1  (git://github.com/finventures/chrome-webstore-upload-cli.git#788fc45c856bf5f5befd2aa3c2e4e000314eb651)
  └── chrome-webstore-upload@0.2.1  (git://github.com/drewml/chrome-webstore-upload.git#ee0a1910f6318098df132f4eeb3509b6ae8a49b9)

root@ubuntu:~/myproject# vi manifest.json 
root@ubuntu:~/myproject# webstore upload --extension-id nnoooooooooooooooooooooooooooooo --trusted-testers --auto-publish
Response code 400 (Bad Request)
{
    "error": {
        "errors": [
            {
                "domain": "global",
                "reason": "badRequest",
                "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
            }
        ],
        "code": 400,
        "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
    }
}
```

**Comment by p0we7** [comment]
### More Error Information

```
root@ubuntu:~/v2ex-plus# webstore upload --extension-id nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn --trusted-testers
Upload Completed

root@ubuntu:~/v2ex-plus# webstore publish --extension-id nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn --trusted-testers
Response code 400 (Bad Request)
{
    "error": {
        "errors": [
            {
                "domain": "global",
                "reason": "parseError",
                "message": "This API does not support parsing form-encoded input."
            }
        ],
        "code": 400,
        "message": "This API does not support parsing form-encoded input."
    }
}
root@ubuntu:~/v2ex-plus# webstore publish --extension-id nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn 
Response code 400 (Bad Request)
{
    "error": {
        "errors": [
            {
                "domain": "global",
                "reason": "badRequest",
                "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
            }
        ],
        "code": 400,
        "message": "Publish condition not met: This item is current published to trusted testers. To publish it to everyone, you must unpublish it first. To keep publishing it to trusted testers, set publish_to_trusted_testers to true in the request."
    }
}
root@ubuntu:~/v2ex-plus# 
```

**Comment by bfred-it** [comment]
Are you updating the version? Maybe you need to increase it first

**Comment by p0we7** [comment]
@bfred-it 

```
root@ubuntu:~/v2ex-plus# npm ls -g
/usr/lib
├─┬ chrome-webstore-upload-cli@1.1.1 (git://github.com/finventures/chrome-webstore-upload-cli.git#788fc45c856bf5f5befd2aa3c2e4e000314eb651)
│ ├─┬ chalk@1.1.3
│ │ ├── ansi-styles@2.2.1
│ │ ├── escape-string-regexp@1.0.5
│ │ ├─┬ has-ansi@2.0.0
│ │ │ └── ansi-regex@2.1.1
│ │ ├── strip-ansi@3.0.1
│ │ └── supports-color@2.0.0
│ ├─┬ chrome-webstore-upload@0.2.1 (git://github.com/drewml/chrome-webstore-upload.git#ee0a1910f6318098df132f4eeb3509b6ae8a49b9)
│ │ └─┬ got@6.7.1
```

**Comment by bfred-it** [comment]
No I mean your extension’s version. Maybe the API says that the current version is *already* published to trustedTesters and the only possible action is to publish it to everyone

**Comment by hkdobrev** [comment]
@bfred-it Do you think you could merge and release this branch? https://github.com/DrewML/chrome-webstore-upload/compare/trusted_testers

**Comment by bfred-it** [comment]
@hkdobrev I would if I have confirmation that it works. @p0we7 mentioned it doesn't work for him.

**Comment by hkdobrev** [comment]
@squidarth @bfred-it I was able to successfully publish my extension to trusted testers using the request format in the master branch of the client (using just query param `publishTarget=trustedTesters`).

My confusion was created from a bug here which didn't send the target when using `webstore upload --auto-publish --trusted-testers`. I've created a PR targetting this PR at https://github.com/finventures/chrome-webstore-upload-cli/pull/1

Could you please test/merge/release that? Thank you! Much appreciated!

**Comment by bfred-it** [comment]
Whoever can test this, install the version in this branch with:

```sh
npm rm -g chrome-webstore-upload-cli && npm i -g finventures/chrome-webstore-upload-cli
```

I'll merge if you confirm it works and then later @DrewML can publish it to npm (unless he adds me on npm)

**Comment by gcooney** [comment]
@bfred-it Not a committer but I was able to use this to deploy an extension with to set of trusted testers.

**Comment by greghuc** [comment]
@bfred-it just tested finventures/chrome-webstore-upload-cli, and successfully deployed an extension to a set of trusted testers.

Would love to re this PR be merged.

**Comment by hkdobrev** [comment]
@DrewML I've been using the fork for months successfully with both unlisted and private extension with trusted testers. It works fine in all cases I had. Do you think we could merge this? Thank you! ❤️

**Comment by bfred-it** [comment]
Thanks for the reports! I merged this, however I don't have npm rights to publish it. I hope @DrewML will do so soon :)

**Comment by greghuc** [comment]
Hi @DrewML could you please push the latest version of chrome-webstore-upload-cli to npm please? The latest code (committed 3 months ago) adds support for "trustedTesters", which is great. Many thanks.

**Comment by RichardIvan** [comment]
Hi @DrewML, would it be possible to release it to the npm please?

**Comment by chrmod** [comment]
We have released a version of chrome-webstore-upload-cli under our organization https://www.npmjs.com/package/@cliqz-oss/chrome-webstore-upload-cli so fill free to use it (published code is just master from uptream repo)

**Comment by bfred-it** [comment]
Sorry for the delay, I just realized that I also have npm publish rights now.

```
❯ np minor
 ✔ Prerequisite check
 ✔ Git
 ✔ Cleanup
 ✔ Installing dependencies using npm
 ✔ Running tests using npm
 ✔ Bumping version using npm
 ✔ Publishing package using npm
 ✔ Pushing tags

 chrome-webstore-upload-cli 1.2.0 published 🎉
```

**Pull request merged**